### PR TITLE
Use hostprocess for Windows node manager in Helm chart

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it: 

Currently, external cloud-provider does not work with Windows nodes when using the Helm chart. The Daemonset for cloud-node-manager-windows is not able to reach the metadata endpoint and therefore windows nodes always have the Taint node.cloudprovider.kubernetes.io/uninitialized=true:NoSchedule.

This PR changes the cloud-node-manager-windows Daemonset to use `hostProcess: true` and adds a temporary workaround for not being able to use `inClusterConfig()` (until k8s 1.26 + containerd v1.7).

All credits go to @mweibel and @marosset for discovering the issue and suggesting a fix.

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2591

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: helm/cloud-provider-azure/templates/cloud-provider-azure.yaml
Comment (CecileRobertMichon) [inline helm/cloud-provider-azure/templates/cloud-provider-azure.yaml:12]: will rebase once #3281 merges

Comment (CecileRobertMichon) [reply helm/cloud-provider-azure/templates/cloud-provider-azure.yaml]: done

**Comment by feiskyer** [comment]
> the Daemonset for cloud-node-manager-windows is not able to reach the metadata endpoint 

Could you elaborate why IMDS is not reachable on capz?

File: helm/cloud-provider-azure/templates/cloud-provider-azure.yaml
Comment (feiskyer) [inline helm/cloud-provider-azure/templates/cloud-provider-azure.yaml:442]: Could the in-cluster authz with service account work here? If so, there is no need of the script

Comment (marosset) [reply helm/cloud-provider-azure/templates/cloud-provider-azure.yaml:442]: In-cluster auth with service accounts works for hostprocess containers with containerd v1.7 only (not v1.6.x).
The way volume mounts get set up was changed between containerd 1.6 and 1.7

Comment (CecileRobertMichon) [reply helm/cloud-provider-azure/templates/cloud-provider-azure.yaml:442]: yes this is a workaround until containerd 1.7 is released, after that we can remove the script

**Comment by CecileRobertMichon** [comment]
> Could you elaborate why IMDS is not reachable on capz?

@marosset can weigh in but I believe it's not a capz limitation, it's a calico overlay networking issue: https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2132

Azure CSI driver helm charts also had to do this for this reason: https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/1201

**Review by marosset — APPROVED** [review]
/lgtm

This should unblock deploying the Windows cloud-node-manager!
We can update the deployment to no longer need the powershell / kube-config after containerd 1.7 releases (but the current changes should work on both containerd 1.6 and 1.7)

File: helm/cloud-provider-azure/templates/cloud-provider-azure.yaml
Comment (feiskyer) [reply helm/cloud-provider-azure/templates/cloud-provider-azure.yaml:442]: then let's wait for 1.7?

Comment (CecileRobertMichon) [reply helm/cloud-provider-azure/templates/cloud-provider-azure.yaml:442]: it's blocking https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3105 

we want to move all CAPZ templates and tests to external cloud-provider to get as much test coverage as possible before in-tree cloud-providers are completely removed

Comment (feiskyer) [reply helm/cloud-provider-azure/templates/cloud-provider-azure.yaml:442]: I see. Could you add a TODO comment with containerd issue link?

Comment (marosset) [reply helm/cloud-provider-azure/templates/cloud-provider-azure.yaml:442]: Here is the issue https://github.com/kubernetes/kubernetes/issues/104562

Comment (CecileRobertMichon) [reply helm/cloud-provider-azure/templates/cloud-provider-azure.yaml:444]: Added

**Review by feiskyer — APPROVED** [review]
/lgtm

**Pull request merged**
